tests/functional/builtins/codegen/test_uint2str.py

```diff
@@ -37,3 +39,15 @@ def foo(x: uint{bits}) -> uint256:
     """
     c = get_contract(code)
     assert c.foo(2**bits - 1) == 0, bits
+
+
+def test_int_fails(get_contract):
+    code = """
+@external
+def test():
+    a: String[78] = uint2str(-1)
+    pass
+    """
+
+    with pytest.raises(InvalidType):
+        get_contract(code)
```

Comment:
i'd prefer `compile_code` here, it has fewer side effects

Reply:
Makes sense, thank you!